bson/bsoncodec/slice_codec.go

```diff
@@ -23,7 +23,7 @@ type SliceCodec struct {
 	EncodeNilAsEmpty bool
 }
 
-var _ ValueCodec = &MapCodec{}
+var _ valueCodec = &MapCodec{}
```

Comment:
Could we add a comment describing the compile check?

Reply:
This compile-time assertion is actually redundant because the codec is already passed into `RegisterTypeEncoder` and `RegisterTypeDecoder`, so I will remove it.